tests/test_panther_python_backend.py

```diff
@@ -0,0 +1,418 @@
+    if re.match("^[Cc]:\\[Pp]rogram[Dd]ata\\.{1,9}\\.exe", event.deep_get("ImagePath", default="")):
+        return True
+    return False
+"""
+
+    result = convert_rule(rule)
+    assert result == expected_result
+
+
+def test_condition_endswith(backend):
+    rule = """
+    selection:
+        fieldA|endswith: valueA
+    condition: selection
+    """
+
+    expected_result = """def rule(event):
+    if event.deep_get("fieldA", default="").endswith("valueA"):
+        return True
+    return False
+"""
+
+    result = convert_rule(rule)
+    assert result == expected_result
+
+
+def test_condition_startswith(backend):
+    rule = """
+    selection:
+        fieldA|startswith: valueA
+    condition: selection
+    """
+
+    expected_result = """def rule(event):
+    if event.deep_get("fieldA", default="").startswith("valueA"):
+        return True
+    return False
+"""
+
+    result = convert_rule(rule)
+    assert result == expected_result
+
+
+def test_condition_contains(backend):
+    rule = """
+    selection:
+        fieldA|contains: valueA
+    condition: selection
+    """
+
+    expected_result = """def rule(event):
+    if "valueA" in event.deep_get("fieldA", default=""):
+        return True
+    return False
+"""
+
+    result = convert_rule(rule)
+    assert result == expected_result
+
+
+def test_condition_contains_with_backslash(backend):
+    rule = """
+    selection:
+        fieldA|contains: \\valueA\\valueB
+    condition: selection
+    """
+
+    expected_result = """def rule(event):
+    if "\\\\valueA\\\\valueB" in event.deep_get("fieldA", default=""):
+        return True
+    return False
+"""
+
+    result = convert_rule(rule)
+    assert result == expected_result
+
+
+def test_selection_and_not_filter(backend):
+    rule = """
+    selection:
+        fieldA|contains: valueA
+    filter:
+        Image:
+            - 'qrs'
+            - 'xyz'
+        OriginalFileName:
+            - 'abc'
+            - 'efg'
+    condition: selection and not filter
+    """
+
+    expected_result = """def rule(event):
+    if all(
+        [
+            "valueA" in event.deep_get("fieldA", default=""),
+            not any(
+                [
+                    event.deep_get("Image", default="") in ["qrs", "xyz"],
+                    event.deep_get("OriginalFileName", default="") in ["abc", "efg"],
+                ]
+            ),
+        ]
+    ):
+        return True
+    return False
+"""
+
+    result = convert_rule(rule)
+    assert result == expected_result
+
+
+def test_selection_or_not_filter(backend):
+    rule = """
+    selection:
+        fieldA|contains: valueA
+    filter:
+        Image:
+            - 'qrs'
+            - 'xyz'
+        OriginalFileName:
+            - 'abc'
+            - 'efg'
+    condition: selection or not filter
+    """
+
+    expected_result = """def rule(event):
+    if any(
+        [
+            "valueA" in event.deep_get("fieldA", default=""),
+            not any(
```

Comment:
This is not correct, should be `not all(`

Reply:
Why? Doesn't `not filter` mean that no filter conditions should be met?